packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -528,7 +571,15 @@ const getModel = (): Record<string, any> => {
     unsetNullForeignKey(fieldNameDotNotation, outputModel)
   })
 
-  return unFlattenObject(outputModel)
+  const model: Record<string, any> = unFlattenObject(outputModel)
+
+  // Handle the special case of the freeform plugin
+  if (freeformName.value) {
+    const configSchema = props.rawSchema?.fields?.find((field: any) => 'config' in field)?.config
+    model.config = pruneRecord(freeformConfig.value, configSchema)
```

Comment:
As we are not building the model from the flattened schema, we'll need to prune out undefined fields in the raw schema instead.